yarn-project/archiver/src/archiver/archiver.ts

```diff
@@ -266,17 +310,17 @@ export class Archiver implements ArchiveSource {
     }
 
     // create the block number -> block hash mapping to ensure we retrieve the appropriate events
-    const blockNumberToBodyHash: { [key: number]: Buffer | undefined } = {};
+    const blockHashMapping: { [key: number]: Buffer | undefined } = {};
     retrievedBlocks.retrievedData.forEach((block: L2Block) => {
-      blockNumberToBodyHash[block.number] = block.body.getCalldataHash();
+      blockHashMapping[block.number] = block.header.bodyHash;
     });
     const retrievedContracts = await retrieveNewContractData(
       this.publicClient,
       this.contractDeploymentEmitterAddress,
       blockUntilSynced,
       lastL1Blocks.addedBlock + 1n,
       currentL1BlockNumber,
-      blockNumberToBodyHash,
+      blockHashMapping,
     );
 
     this.log(`Retrieved ${retrievedBlocks.retrievedData.length} block(s) from chain`);
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: line 370]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->Optimization wise for storage purposes. It seems useful (in separate pr) to not store the full blocks like this, but have the bodies stored and the headers, then you can always join them to have the full block whenever needed but we don't store anything redundantly and gives us nice flexibility for fetching what we really need. <!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/4632?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->